glusterd2/volgen2/store-utils.go

```diff
@@ -46,5 +58,34 @@ func GetVolfile(volfileID string) ([]byte, error) {
 	if len(resp.Kvs) == 0 {
 		return []byte{}, errors.ErrVolFileNotFound
 	}
-	return resp.Kvs[0].Value, nil
+	var v volfileData
+	if e = json.Unmarshal(resp.Kvs[0].Value, &v); e != nil {
+		log.WithError(e).Error("Failed to unmarshal the data into volfiledata object")
+		return nil, e
+	}
+	return []byte(v.Content), nil
+}
+
+// DeleteVolumeVolfiles deletes all the Volfiles belonging to a Volume
+func DeleteVolumeVolfiles(volname string) error {
+	resp, e := store.Get(context.TODO(), volfilePrefix, clientv3.WithPrefix())
```

Comment:
Snapshot client volfiles will be under volfilePrefix/snaps/<volname>.  Does store.Get returns the files by recursive crawling?

Reply:
Mostly yes. I will verify this use case.

Reply:
`clientv3.WithPrefix()` will include all keys with the prefix

Reply:
Suggestion: If all volfiles belonging to a volume are under one prefix - `volfiles/<volname>, you can delete all with one op using `WithPrefix()`